src/VecSim/vec_sim_interface.h

```diff
@@ -69,6 +72,18 @@ struct VecSimIndexInterface : public VecsimBaseObject {
      */
     virtual size_t indexSize() const = 0;
 
+    /**
+     * @brief Return the index capacity, so we know if resize is required for adding new vectors.
+     *
+     * @return index capacity.
+     */
+    virtual size_t indexCapacity() const = 0;
+
+    /**
+     * @brief Change the index capacity (without chaning its data), by adding another block.
+     */
+    virtual void resize() = 0;
```

Comment:
`resize` is ambiguous 
consider a name like `resizeAddBlock`

Reply:
How about `increaseCapacity`?